module-3/lab-supervised-learning-feature-extraction/your-code/main.ipynb

```diff
@@ -12,7 +12,7 @@
   },
```

Comment:
- Your code doesn't check if the M is in the last character just if M is in the string . This time it works because you only have this character otherwise it wouldn't work. You can do something like the following:
```Python
#Your code 
def convert_string_to_numeric(s):
    """
    Convert a string value to numeric. If the last character of the string is `M`, obtain the 
    numeric part of the string, multiply it with 1,000,000, then return the result. Otherwise, 
    convert the string to numeric value and return the result.
    
    Args:
        s: The Reviews score in string format.

    Returns:
        The correct numeric value of the Reviews score.
    """
    if "M" in s:
        return int(s.split("M")[0][:-2]) * 1000000 
    else:
        return float(s)    
test_string = '4.0M'
convert_string_to_numeric(test_string) == 4000000
#A possible way of doing it 
def convert_string_to_numeric(s):
    if s[-1] == 'M':
        return float(s[:-1]) * 1000000
    else:
        return float(s)
    return np.NaN
test_string = '4.0M'
convert_string_to_numeric(test_string) == 4000000
````
- In this one " While we can convert most of the Size values to numeric in the same way we converted the Reviews values, there is one value that is impossible to convert.
What is that badass value? Enter it in the next cell and calculate the proportion of its occurence to the total number of records of google_play." you should calculate a proportion. I think you've missed something in your code:
```Python 
#your code 
odd_value = "Varies with device"
google_play[google_play.Size == odd_value].shape
#Doing something like the following you will get a proportion
odd_value = "Varies with device"
google_play[google_play.Size == odd_value].shape[0] / google_play.shape[0]
```